### PR TITLE
eigenvalue calculation with arpack and parpack

### PR DESCRIPTION
**Pull request by asartori86**

### CODE REVIEW

File: source/base_interface.cc [hunk 1]
```diff
@@ -405,6 +406,38 @@ set_stepper (const std::string &s) const
...
+
+  for (unsigned int q=0; q<nq; ++q)
+    for (unsigned int c=0; c<n_components; ++c)
+      for (unsigned int i=0; i<dofs_per_cell; ++i)
+        for (unsigned int j=0; j<dofs_per_cell; ++j)
+          data.local_matrix(i,j) +=
+            fev.shape_value_component(i,q,c)
+            *
+            fev.shape_value_component(j,q,c)
+            *fev.JxW(q);
```
Comment (luca-heltai) [inline source/base_interface.cc]: Suboptimal... maybe use `fe.system_to_component_index(i).first` if `fe->is_primitive()`

**Comment by asartori86** [comment]
@luca-heltai -- serial version works quite well. The parallel one not. To my understanding, even if it is templated, it doesn't handles **Block** stuff

**Comment by asartori86** [comment]
@luca-heltai --- as pointed out by @gpitton https://trilinos.org/packages/nox-and-loca/

**Comment by asartori86** [comment]
eig0

**Comment by asartori86** [comment]
eig1

**Comment by asartori86** [comment]
eig2

**Comment by asartori86** [comment]
eig3

**Comment by asartori86** [comment]
eig4

**Comment by asartori86** [comment]
eig5

**Comment by asartori86** [comment]
eig6

**Comment by asartori86** [comment]
eig7

**Comment by asartori86** [comment]
eig8

**Comment by asartori86** [comment]
eig9

**Comment by asartori86** [comment]
eig10

**Comment by asartori86** [comment]
eig11

File: source/base_interface.cc [hunk 2]
```diff
@@ -397,6 +398,42 @@ output_solution (const unsigned int &current_cycle,
...
+      for (auto &name : eig_names)
+        {
+          name += "_";
+          name += Utilities::int_to_string(i);
+        }
+
+      data_out.add_data_vector (eigenvectors[i],
+                                get_component_names());
+
+      data_out.write_data_and_clear(get_mapping());
```
Comment (luca-heltai) [inline source/base_interface.cc:432]: why don't you write all eigenvectors on a single file, instead of writing one file for eigenvector?

Comment (asartori86) [inline source/base_interface.cc:432]: If I put everything in one file is (with paraview) more difficult to switch among eigenvectors. especially if I have more variables. This way the different eigenvectors are seen as different time step.

Tell me what do you prefear

**Comment by asartori86** [comment]
PR https://github.com/dealii/dealii/pull/2617 is required

**Comment by asartori86** [comment]
@luca-heltai @ESeNonFossiIo @nicola-giuliani this is ready for review

**Comment by asartori86** [comment]
I merge with the skype approval of @luca-heltai  :)

**Pull request merged**